[CS2113T-T09-3] HalpMi

tanweili opened the pull request:
The hospital management system is a CLI application that helps to manage all hospital data with ease and efficiency where the user is the hospital administrator.

shxr3f committed:
Merge pull request #132 from tanweili/WeiLi-UpdateDG_UML

Add UML for Med and Appt classes

shxr3f committed:
Making corrections to format

shxr3f committed:
Merge branch 'master' into branch-DG_Corrections

# Conflicts:
#	docs/DeveloperGuide.md

shxr3f committed:
More changes to formatting

shxr3f committed:
Adding full stops.

tanweili committed:
Merge pull request #135 from shxr3f/branch-DG_Corrections

Correcting Format Issues present in DG

wraineflores committed:
DG linking png images

wraineflores committed:
Merge pull request #136 from wraineflores/wraineflores-update-v3

DG linking png images

GlendonNotGlen commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
-## Non-Functional Requirements
+### Use cases
 
-{Give non-functional requirements}
 
+### Non-Functional Requirements
+Device Environment:
+* Must have Java 11 or higher installed in OS
+* 32-bit or 64-bit environment
+* Command Line Interface supported
+
+  Performance of app:
+* Function offline, without the need for internet access
+* Quick to launch and use
+* No noticeable lag or delay in performance when running
+* Intuitive and seamless for new users.
+* Ability to export the data into a txt file to load on another OS
+
+  Reliability of app:
+* Data files should be updated constantly and accurately, with no data loss
+* Data records should be retrievable and readable
+* Text inputs should produce similar results if utilised multiple times.
+* Program should run without any forced-close error due to bugs
+--------------------------------------------------------------------------------------------------------------------
 ## Glossary
+* *FUllNAME* - Standard form for fullname of patients and doctors is a String value with no spaces
+* *NRIC* - Standard form for nric of patients and doctors is a String value with no spaces
+* *AGE* - Standard form for age is an int value more than 0
+* *GENDER* - Standard form for gender of patients and doctors is a char value of "M" or "F"
+* *ADDRESS* - Standard form for address is a String value with no spaces
+* *DOB* - Standard form for date-of-birth is a String value with no spaces
+* *SPECIALIZATION* - Standard form for specialization of doctors is String value with no spaces
+* *MEDICINEID* - Standard form for medicine id is a String value with no spaces
+* *MEDICINENAME* - Standard form for medicine name is a String value with no spaces
+* *DOSAGE* - Standard form for dosage of medicine is an int value, standard unit milligrams
+* *EXPIRY* - Standard form for expiry of medicine is a String value with no spaces
+* *SIDEEFFECTS* - Standard form for side effects of medicine is a String value with no spaces
+* *QUANTITY* - Standard form for quantity of medicine is an int value
+* *APPOINTMENTID* - Standard form for appointment id is a String value with no spaces
+* *APPOINTMENTDETAILS* - Standard form for appointment details is a String value
```
Great use of a glossary here! Really helps in understanding the DG better

GlendonNotGlen commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+The rest of the App consists of these components.
+
+* [**`Manager`**](#manager-component): The Brain.
```
Not sure if calling The Brain would be appropriate? Can perhaps change it to main driver function or say 'where most of the logic is computed'?

ngys117 replied:
I'm not sure if this description is helpful. Maybe be a bit more explicit with what the Manager actually does? From the diagram it seems like it acts as an intermediary between the User/Duke and the rest of the components.

Ch40gRv1-Mu replied:
Confusing expression. Explain what does this component do specific instead.

GlendonNotGlen commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+### Architecture
+
+![Architecture Diagram](diagrams/OverallArch.png)
```
Really great diagram that gives an overall architecture overview of your program

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+#### `Validator`
+
+![ValidatorClassUML](diagrams/ValidatorClassUML.png)
```
Perhaps the color is too dark, making the black text hard to read.

Ch40gRv1-Mu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+### Architecture
+
+![Architecture Diagram](diagrams/OverallArch.png)
+
```
Why is there a component named Duke in the overall architecture while your product name is not Duke. Do you mean the main function?

Ch40gRv1-Mu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+#### **Main components of the architecture**
+
+**`Duke`** has single method called `main` which is called upon launch. This initialises a new instance of a `Manager` class, and calls the `runLoop()`
+method belonging to the Manager object.
```
Suggests to change the name of "Duke" to "Main" instead to avoid confuseness.

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+#### `Storage`
+
+![StorageClassUML](diagrams/StorageClassUML.png)
```
Maybe there are too many inheritance arrowheads and association arrows overlapping one another?

![image](https://user-images.githubusercontent.com/70083643/160965867-780a72df-b029-45da-ab2d-5984c0ab96b9.png)

Consider changing the formatting so that the either the Storage or List class is at the bottom to remove the overlaps.

GlendonNotGlen commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+### User stories
 
-## User Stories
+Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unlikely to have) - `*`
```
I think the formatting of this legend can be neater? Overall great idea nonetheless!

GlendonNotGlen commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
 
-{Give non-functional requirements}
 
+### Non-Functional Requirements
+Device Environment:
+* Must have Java 11 or higher installed in OS
+* 32-bit or 64-bit environment
+* Command Line Interface supported
+
+  Performance of app:
+* Function offline, without the need for internet access
+* Quick to launch and use
+* No noticeable lag or delay in performance when running
+* Intuitive and seamless for new users.
+* Ability to export the data into a txt file to load on another OS
+
+  Reliability of app:
+* Data files should be updated constantly and accurately, with no data loss
+* Data records should be retrievable and readable
+* Text inputs should produce similar results if utilised multiple times.
+* Program should run without any forced-close error due to bugs
+--------------------------------------------------------------------------------------------------------------------
```
Do becareful of formatting error. The sub headers here seem to be misaligned

ngys117 replied:
Are the indentations for "Performance of app" and "Reliability of app" here intentional? Looks inconsistent with the "Device Environment" portion.

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
-### Target user profile
 
-{Describe the target user profile}
+###Target user profile:
```
Seems like the header formatting is showing in the actual webpage?
![image](https://user-images.githubusercontent.com/70083643/160966007-06ffdc4f-aeb8-48d3-a52a-901ce0813ccf.png)

Ch40gRv1-Mu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+* [**`Manager`**](#manager-component): The Brain.
+* [**`Helper`**](#helper-classes): A collection of core classes that aid with the Logical Operations performed by HalpMi.
+    * [**`Command`**](#command): Contains the changes or updates to be made.
+    * [**`UI`**](#ui-component): The UI of the App.
+    * [**`Parser`**](#parser): Breaks down user input into parameters accepted by the app and creates a Command Object.
+    * [**`Validator`**](): Checks if the input provided by the User is Valid.
+    * [**`Storage`**](#storage): Reads data from data files and writes data to data files, also stores in app memory.
+
```
Is this correct indentation? or Command, UI, Parser, Validator, Storage belongs to Helper? which does not make sense

GlendonNotGlen reviewed (commented):
Great implementation of DG overall. The DG was easy to read and diagrams are very neat without being overly cluttered.

Ch40gRv1-Mu commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+of `Appointment` as a private element. `AppointmentList` methods invoked interacts with this
+list, possibly making changes in the process.
+
+Currently, `AppointmentList` has the following methods:
+* `AppointmentList#add` -- Appends a new `Appointment` to the list.
+* `AppointmentList#remove` -- Removes an entry from the list by appointment id.
+* `AppointmentList#edit` -- Edits a existing entry from the list by.
+* `AppointmentList#view` -- Text display of all or selected appointments in the list.
+* `AppointmentList#find` -- Find selected appointments using the criteria given.
```
May hide some unimportant detail but only state the noteworthy part instead.

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+* prefers typing to mouse interactions
+* is reasonably comfortable using CLI apps
+
+###Value proposition:
```
Same issue with header formatting here.

limjierui commented on docs/diagrams/SequenceDiagram.puml:
```diff
@@ -0,0 +1,111 @@
+deactivate Duke
+destroy Duke
+
+@enduml
```
Understand that there are many method calls in this sequence diagram, however the sequence diagram looks like there is alot going on. Perhaps it would be better to break it down into separate sequence diagrams and reference each other.

Ch40gRv1-Mu commented on docs/diagrams/AppointmentListUML.puml:
```diff
@@ -0,0 +1,22 @@
+@startuml
+
+class AppointmentList {
+-appointments : Arraylist<appointment> = null
+-returnedFinderArray : Arraylist<appointment> = null
++getAppointment(String) : Appointment
++getList() : Arraylist<appointment>
++add(String[]) : void
++remove(String) : void
++edit(String[]) : void
++view() : void
++view(String) : void
```
Should we follow the notation taught in class? e.g. + for public, - for private etc.

limjierui commented on docs/diagrams/ValidatorClassUML.puml:
```diff
@@ -0,0 +1,33 @@
+-validateQuantity(String): boolean
+-validateSpecialization(String): void
+}
+@enduml
```
This class seem to contain alot of methods and attributes, and it looks lengthy. Perhaps it be better to include only important attributes and methods.

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+The Sequence Diagram below showcases the general Logic and Flow of the program from Launch till Exit.
+
+![Sequence Diagram](diagrams/SequenceDiagram.png)
```
Not sure if the sequence diagram should have additional objects at the footing.

FaliciaOng replied:
![image](https://user-images.githubusercontent.com/69444353/160971296-e8ce44f8-9350-4e2a-a4a7-c407b028ddce.png)

Is this format correct? Should the lifeline stop after the deletion of an object?

Ch40gRv1-Mu commented on docs/diagrams/CommandUML.puml:
```diff
@@ -0,0 +1,32 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+alt add patient
+    Manager -> Command: addPatient
+    Command -> PatientList: add
+else delete patient
+    Manager -> Command: deletePatient
```
Suggest to add a notation to declare that this is  a minimum notation sequence diagram.

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+#### `Parser`
+
+![ParserClassUML](diagrams/ParserClassUML.png)
```
It seems like the parser methods are static. Perhaps they should be underlined in the class diagram?

Ch40gRv1-Mu commented on docs/diagrams/SequenceDiagram.puml:
```diff
@@ -0,0 +1,111 @@
+User -> Duke : Launches App
+activate Duke
+
+Duke -> Manager : Manager()
+activate Manager
```
Is this wrong notation for execution of constructor? The activation bar should jus under the class. You may miss "create Manager"

Ch40gRv1-Mu commented on docs/diagrams/SequenceDiagram.puml:
```diff
@@ -0,0 +1,111 @@
+Duke -> Manager : Manager()
+activate Manager
+
+Manager -> UI: UI()
+activate UI
+UI --> Manager
```
Same as the previous comments

Ch40gRv1-Mu commented on docs/diagrams/SequenceDiagram.puml:
```diff
@@ -0,0 +1,111 @@
+Manager --> Duke
+deactivate Manager
+destroy Assets
+destroy Storage
+destroy UI
+destroy Parser
+destroy Validator
+destroy Manager
+deactivate Duke
+destroy Duke
+
```
So review all the execution of construction pls.

Ch40gRv1-Mu commented on docs/diagrams/StorageClassUML.puml:
```diff
@@ -0,0 +1,37 @@
+class MedicineList
+class AppointmentList
+
+List <|-- PatientList
+List <|-- DoctorList
+List <|-- MedicineList
+List <|-- AppointmentList
```
The arrows are too messy. Moreover, I didn't see the value to express the association between Storage and all specific XYZList, which doesn't add value. Can just show the a specific case as example

limjierui commented on docs/diagrams/CommandUML.puml:
```diff
@@ -0,0 +1,32 @@
+    Manager -> Command: viewMedicine
+    Command -> MedicineList: viewMedicine
+end
+@enduml
```
Would any of the commands return any output or list? If it does, will be good to show a return arrow back to the main class that calls the commands.

Ch40gRv1-Mu reviewed (commented):
Some diagrams are too complex and hard to read for reader. May suggest to hide some details. For example, there's no need to show all the methods of classes in the Class diagram? (Can use notation to inform you hide some unimportant methods instead)

limjierui commented on docs/diagrams/ParserClassUML.puml:
```diff
@@ -0,0 +1,31 @@
++parseViewMedicine(String): Command
++parseViewPatient(String): Command
+}
+@enduml
```
Again, there seems to be too many methods displayed in this function, perhaps it will be better to reduce and display only important methods and attributes.

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+For example, validateAddPatient validates the parameter of `add patient` command, ensuring each parameter is in correct
+format. Please refer to the below sequence diagram for a clearer understanding.
+
+![ValidatorUML](diagrams/ValidatorUML.png)
```
It seems like the validator methods are static as well. Perhaps they should be underlined in the class diagram?

FaliciaOng replied:
![image](https://user-images.githubusercontent.com/69444353/160971864-de63dcf5-5645-44ac-99da-1b6d5bb2ba30.png)

Should there be a self-invocation bar for each calling of own method?

ngys117 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,324 @@
+
+### UI component
+
+![UIClassUML](diagrams/UIClassUML.png)
```
Consider following the lecture uml diagram notation for showing visibility and abstract classes.

ngys117 reviewed (commented):
Mostly seems fine. Just need to take note of static members/methods in class diagrams. Another comment is maybe consider following the lecture uml diagram notation for showing visibility and abstract classes.

FaliciaOng reviewed (commented):
The developer guide was fairly clear. Good job!

shxr3f committed:
Changing saveString() method for patient and doctor

tanweili committed:
Merge pull request #137 from shxr3f/branch-correctBugs

Changing saveString() method for patient and doctor

tanweili committed:
Implement IdGenerator and changing add appt cmd

IdGenerator has a static method to generator a pseudo unique ID which
is a 8 digit number in String format. Can be used for other lists as
well. Still in the midst of changing the add appt cmd. Currently, it
should check if patient and doctor exists. However, still need to
implement adding into the appointment list.

tanweili committed:
Implement reference from ApptList to PatientList and DtrList

In order to retrieve info regarding patient and doctor names to
simplify add appt command, apptList needs to know about patients
and doctors.

tanweili committed:
Update IdGenerator and implement new add appointment

IdGenerator now has a method that is specific to add appt. New
add appt generates a pseudo unique hash from patient and doctor nric
and appt date.

tanweili committed:
Update appt saveString

So that appt data is correctly loaded upon start up.

tanweili committed:
Update U/DG to reflect changes made to ApptList

tanweili committed:
Fix #138

Made error message for name error to be more explicit in rejecting
special characters and numbers.

tanweili committed:
Fix #139

Made error message for invalid age input to be more explicit in
referring to the age parameter.

tanweili committed:
Fix #141

Update UG DG to reflect the change that view doctor can now view by
specific criteria.

tanweili committed:
Fix #143

Update UG DG to reflect changes to find patient being able to search
by criteria.

tanweili committed:
Fix #148

Update UG DG to reflect changes to view medicine feature.

Anvitha-r committed:
Fix Example documentation for Find commands

Anvitha-r committed:
Fix view appointment examples in user guide

Anvitha-r committed:
Fix command to exit the application not documented and add help fucntion

Anvitha-r committed:
Fix command summary to include edit function

Anvitha-r committed:
fix appointment commands in ug

Anvitha-r committed:
fix indentation errors

Anvitha-r committed:
fix view command summary fucntions for medicine, doctor and patient

Anvitha-r committed:
fix medicine side effects so that confusion doesn't occur

shxr3f committed:
Merge pull request #178 from tanweili/WeiLi-CheckIfPatientAndDoctorExistForAppt_ChangeApptInputs

Implement IdGenerator and changing add appt cmd

wraineflores committed:
update ug delete appointment

wraineflores committed:
update command class uml

Anvitha-r committed:
fix bug to ensure that the age and dob match

wraineflores committed:
updated DoctorList & PatientList method names

wraineflores committed:
update refactor validator find switch cases

wraineflores committed:
update validator

wraineflores committed:
update printAddDoctor message

wraineflores committed:
update print addPatient & addMedicine messages

wraineflores committed:
update view help feature

wraineflores committed:
update delete help message

wraineflores committed:
update delete help message v2

wraineflores committed:
update edit help message

shxr3f committed:
Implementing a CheckForMedicineStockCommand
on execute it checks of the medicines given as input are available in inventory
prints a table of medicine that are not available

shxr3f committed:
Implementing CheckIfAppointmentTodayCommand
on execute checks if the patient has an appointment today else throws exception

wraineflores committed:
update find & additional features in help

shxr3f committed:
Creating DispenseMedicineCommand
the execute method updates the quantitiy of medicines in the medicine list
it also updates the medicines dispenses on the appointment object

shxr3f committed:
Implementing new methods in AppointmentList class
these methods assisst in dispensing medicine feature.

shxr3f committed:
Implementing new methods in MedicineList class
these methods assisst in dispensing medicine feature.

shxr3f committed:
Making quantity protected instead of private

shxr3f committed:
New saveMedicineStringMethod
this saves the details of the medicines that was dispensed during the appt

shxr3f committed:
Implementing new parser methods for dispense medicine feature.
Also implemented a new parameter check method that does not check for length

shxr3f committed:
Creating validator method for dispense medicine

shxr3f committed:
Added some new statuses for dispense medicine feature

shxr3f committed:
creating load and save methods to save medicine data in appointments

shxr3f committed:
Included the dispense medicine case in manager class

shxr3f committed:
Refractoring ExpiryComparator class

shxr3f committed:
Resolving checkstyle issues

wraineflores committed:
Merge pull request #182 from wraineflores/wraineflores-update-v5

 Updated help feature in UI v3

tanweili committed:
Create my draft PPP

Implement skeleton for PPP first.

tanweili committed:
Merge pull request #184 from tanweili/WeiLi-PPPDraft

Create my draft PPP

wraineflores committed:
Update appointment results

wraineflores committed:
Change to public

wraineflores committed:
Merge branch 'master' into fix-bugs

wraineflores committed:
Update Validator.java

tanweili committed:
Merge pull request #183 from shxr3f/branch-assignMedsFeature

Dispense Medicine Feature

wraineflores committed:
Merge branch 'master' into wraineflores-update-v5

wraineflores committed:
Merge remote-tracking branch 'upstream1/master' into wraineflores-update-v5

# Conflicts:
#	src/main/java/seedu/duke/assets/Appointment.java
#	src/main/java/seedu/duke/helper/Validator.java

wraineflores committed:
Merge branch 'wraineflores-update-v5' of https://github.com/wraineflores/tp into wraineflores-update-v5

# Conflicts:
#	src/main/java/seedu/duke/helper/Validator.java

wraineflores committed:
Created PPP md file

tanweili committed:
Merge pull request #180 from Anvitha-r/fix-bugs

fix bugs

wraineflores committed:
Fix update appointment bug

wraineflores committed:
Merge remote-tracking branch 'upstream1/master' into wraineflores-update-v5

# Conflicts:
#	docs/UserGuide.md

tanweili committed:
Add med name validation now accepts spaces

Some medicine names have spaces so I think we should accept it as
well.

tanweili committed:
Update add appt details in UG

To reflect the new inputs of add appt command.

shxr3f committed:
Updating PPP

tanweili committed:
Update UG command summary table

add medicine currently still need id so I added that in inside the
table.

Changed add appt input to the new version.

Changed the example id for finding and deleting appt to be a new
appt id example.

wraineflores committed:
Adds PPP.md template for all

wraineflores committed:
Update add appointment & appointment id help message examples

shxr3f committed:
Adding Documentation for Manager Class

shxr3f committed:
Adding documentation for Status Class

shxr3f committed:
Adding Documentation for Command classes part 1.

shxr3f committed:
Correcting checkstyle

tanweili committed:
Update all puml diagrams

To remove all class icons and change method/variable icon to
the normal characters instead of the default icons.

kktai1512 committed:
merging from main

shxr3f committed:
Updating Documentation for Command Classes part 2

kktai1512 committed:
abstract out the load function to a generic one

wraineflores committed:
removed one md

kktai1512 committed:
add save

kktai1512 committed:
refactor exceptions

kktai1512 committed:
add corrupted file handling and abstract exceptions, and change validator-parser logic by moving param number check to validator from parser

kktai1512 committed:
change UI help print add doctor / add patient method name

kktai1512 committed:
change duplicate methods - openlengthcheck in parser

tanweili committed:
Remove ApptListTest

Not adding in relevant tests that are updated to the new ApptList
code removing the file entirely.

kktai1512 committed:
fix checkstyle

shxr3f committed:
Merge pull request #186 from tanweili/WeiLi-MinorUpdates

Wei li minor updates

shxr3f committed:
Merge pull request #185 from wraineflores/wraineflores-update-v5

Update appointment results

Anvitha-r committed:
Merge pull request #187 from shxr3f/branch-docsUpdate-Sharif

Branch docs update sharif

kktai1512 committed:
fix checkstyle

kktai1512 committed:
remove comments

tanweili committed:
Merge branch 'master' into CSV-corrupt-handling

tanweili committed:
Merge pull request #188 from kktai1512/CSV-corrupt-handling

Csv corrupt handling

tanweili committed:
Final fixes for V2.1 via zoom meeting

tanweili committed:
Merge pull request #189 from tanweili/WeiLi-Fixes

Final fixes for V2.1 via zoom meeting

wraineflores committed:
update v12

wraineflores committed:
Merge remote-tracking branch 'upstream1/master' into wraineflores-update-v5

# Conflicts:
#	src/main/java/seedu/duke/helper/Validator.java

wraineflores committed:
delete for loop

kktai1512 committed:
edit doctor prompt fix

kktai1512 committed:
change find methods switch case to lower such that it fits our new upper case requirement change

kktai1512 committed:
lower case param[0] in validator find methods

kktai1512 committed:
make sure admission date is after dob

kktai1512 committed:
fix checkstyle

kktai1512 committed:
fix dob admission validator

wraineflores committed:
Update UG v12

wraineflores committed:
Edit v15

wraineflores committed:
Update v16

wraineflores committed:
Update v17

tanweili committed:
Update my PPP

shxr3f committed:
Merge pull request #192 from tanweili/WeiLi-Updates2

Update my PPP

shxr3f committed:
Merge pull request #191 from wraineflores/wraineflores-update-v5

Update UG for submission

shxr3f committed:
Updating user guide
Inserting details of dispense medicine feature

shxr3f committed:
Removing unwanted comments

shxr3f committed:
Updating Help for dispense medicine

shxr3f committed:
Refactoring code in manager to adhere to code style

shxr3f committed:
Edits to text-ui-tests

tanweili committed:
Update DG implemntation with ViewApptCmd

Add in a simplified seq diag and explanation. Remove design
consideration section since I guess nothing's gonna be there.

tanweili committed:
Update PPP to reflect DG design implementation

tanweili committed:
Update AboutUs.md

Add in my photo and add in our PPP links. Not sure if photo size will
have any issues.

wraineflores committed:
Update md v2

wraineflores committed:
Change md v3

wraineflores committed:
Merge pull request #195 from wraineflores/wraineflores-update-v5

Update md v2

shxr3f committed:
Created new sequence diagram from Dispense Medicine Feature

shxr3f committed:
Image for Dispence Med puml

shxr3f committed:
Updating overall sequence diagram and Storage Class Diagram

shxr3f committed:
Editing description of Storage Class and Adding desription for Dispense Medicine Feature

wraineflores committed:
Update wraineflores.md

wraineflores committed:
Update wraineflores.md

wraineflores committed:
Update UserGuide.md

shxr3f committed:
Merge pull request #194 from tanweili/WeiLi-Updates2

Wei li updates2

Anvitha-r committed:
Merge pull request #193 from shxr3f/branch-CodeReview

Code review

kktai1512 committed:
fix edit patient

kktai1512 committed:
Merge branch 'master' into fix-bugs

kktai1512 committed:
fix patientlist error

shxr3f committed:
Merge pull request #190 from kktai1512/fix-bugs

Fix bugs

kktai1512 committed:
added UGDG for storage and corrupted files

tanweili committed:
Some final fixes for submission...

tanweili committed:
Fix link issue for PPP due to github pages file structure

tanweili committed:
Update user stories section of DG

wraineflores committed:
Update wraineflores.md

wraineflores committed:
Update AboutUs.md

Anvitha-r committed:
merge from master to branch

kktai1512 committed:
PPP update

kktai1512 committed:
PPP update

wraineflores committed:
Update wraineflores.md

kktai1512 committed:
validateMedicine method clean up

tanweili committed:
Update UserGuide

Include warning regarding delete and edit commands.
Fixes #163

kktai1512 committed:
validateMedicine method clean up

tanweili committed:
Add line break between diagram and paragraph

tanweili committed:
Merge pull request #197 from tanweili/WeiLi-Fixes3

Some final fixes for submission...

tanweili committed:
Make user stories table nicer

Attempting to fix the table not showing up also.

tanweili committed:
Add more line break to my PPP to make it look nicer

tanweili committed:
Merge pull request #198 from tanweili/WeiLi-Fixes4

Make user stories table nicer

tanweili committed:
Add line break to properly display user stories table

tanweili committed:
Merge pull request #199 from tanweili/WeiLi-AddLineBreak

Add line break to properly display user stories table

wraineflores committed:
Update DG UML

wraineflores committed:
Merge branch 'master' into wraineflores-update-v5

wraineflores committed:
Merge pull request #200 from wraineflores/wraineflores-update-v5

Update DG UML

tanweili committed:
Update UG to change from admission date to registration date

Update the UG command and example also.

tanweili committed:
Update patient table headers to be consistent

Across all views and find patients

tanweili committed:
Update code to accept registrationdate instead of admissiondate

tanweili committed:
Update UG to remove view appt by specific criteria

To be consistent that all views in that section we just show only
view all.

tanweili committed:
Fix DG Glossary section format overflowing to above

kktai1512 committed:
validateMedicine method clean up

kktai1512 committed:
remove unused validator methods

kktai1512 committed:
update UGDG

Anvitha-r committed:
add images to the UG

Anvitha-r committed:
add images for edit function

Anvitha-r committed:
add images for find function

Anvitha-r committed:
update PPP

kktai1512 committed:
update exceptions DG

kktai1512 committed:
update exceptions DG

kktai1512 committed:
add diagrams

kktai1512 committed:
Merge branch 'master' into documentation-update

kktai1512 committed:
add photo

wraineflores committed:
Update Validator.java

kktai1512 committed:
add photo

wraineflores committed:
Merge pull request #201 from tanweili/WeiLi-FindPatientRegistrationDate

Update UG to change from admission date to registration date

kktai1512 committed:
fix validatedosage method

kktai1512 committed:
fix validateQuantity

kktai1512 committed:
add picture

Anvitha-r committed:
changes

wraineflores committed:
Merge pull request #196 from kktai1512/documentation-update

added UGDG for storage and corrupted files

wraineflores committed:
Update UserGuide.md

Anvitha-r committed:
changes to save data

Anvitha-r committed:
Merge branch 'master' into appointment_1

* master: (33 commits)
  Update UserGuide.md
  add picture
  fix validateQuantity
  fix validatedosage method
  Update Validator.java
  add photo
  add diagrams
  update exceptions DG
  update exceptions DG
  update UGDG
  remove unused validator methods
  validateMedicine method clean up
  Fix DG Glossary section format overflowing to above
  Update UG to remove view appt by specific criteria
  Update code to accept registrationdate instead of admissiondate
  Update patient table headers to be consistent
  Update UG to change from admission date to registration date
  Update DG UML
  Add line break to properly display user stories table
  Add more line break to my PPP to make it look nicer
  ...

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/seedu/duke/assets/AppointmentList.java
#	src/main/java/seedu/duke/assets/PatientList.java

shxr3f committed:
Merge pull request #202 from Anvitha-r/appointment_1

Changes

wraineflores committed:
Update UserGuide.md

Anvitha-r committed:
add screenshot to find appointment

Anvitha-r committed:
add findappointment screenshot

Anvitha-r committed:
Merge branch 'master' into UG-updates

kktai1512 committed:
help-change

kktai1512 committed:
fix findpatient exit error

Anvitha-r committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-3/tp

* 'master' of https://github.com/AY2122S2-CS2113T-T09-3/tp:
  Update UserGuide.md
  changes to save data
  changes
  update PPP
  add images for find function
  add images for edit function
  add images to the UG
  merge from master to branch

# Conflicts:
#	docs/UserGuide.md

tanweili committed:
Set theme jekyll-theme-cayman

kktai1512 committed:
crop picture

kktai1512 committed:
add line break

tanweili committed:
Update appt and apptList UML images to reflect new methods

tanweili committed:
Reduce size of Manager UML image

Anvitha-r committed:
add pics for add function UG

Anvitha-r committed:
update UG with screenshot for add

tanweili committed:
Update pt ptList appt apptList UML Diag to reflect new methods added

Anvitha-r committed:
remove edit appointment

Anvitha-r committed:
add screenshots for all functions

wraineflores committed:
Merge pull request #205 from tanweili/WeiLi-FinalFinalFixes

Update appt and apptList UML images to reflect new methods

tanweili committed:
Merge pull request #203 from Anvitha-r/UG-updates

Ug updates

tanweili committed:
Merge branch 'AY2122S2-CS2113T-T09-3:master' into master

Anvitha-r committed:
add missing images

wraineflores committed:
Update wraineflores.md

Anvitha-r committed:
made changes to the logo and Halpmi start page

Anvitha-r committed:
make changes to the UG commands and sections

wraineflores committed:
Update wraineflores.md

Anvitha-r committed:
fix checkstyle issue

Anvitha-r committed:
reverse changes made

wraineflores committed:
Merge pull request #204 from kktai1512/final-checks

Final checks

Anvitha-r committed:
delete the edit appointment image

shxr3f committed:
Removing edit apppointment function and fixing delete appointment bugs

shxr3f committed:
Merge branch 'master' into branch-fixAppointmentBugs

shxr3f committed:
Inserting image into About Us

Anvitha-r committed:
Merge pull request #207 from shxr3f/branch-fixAppointmentBugs

Fix appointment bugs

Anvitha-r committed:
add photo to about us

wraineflores committed:
Update wraineflores.md

Anvitha-r committed:
fix the about us

Anvitha-r committed:
update ppp

Anvitha-r committed:
Merge branch 'master' into UG-update-changes

shxr3f committed:
Fixing issue with loading data of older date

shxr3f committed:
Merge pull request #208 from shxr3f/branch-fixStorageBugs

Fixing issue with loading data of older date

Anvitha-r committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-3/tp

* 'master' of https://github.com/AY2122S2-CS2113T-T09-3/tp:
  Fixing issue with loading data of older date
  Update wraineflores.md
  Inserting image into About Us
  Removing edit apppointment function and fixing delete appointment bugs
  Update wraineflores.md
  Update wraineflores.md
  add screenshots for all functions
  remove edit appointment
  Update pt ptList appt apptList UML Diag to reflect new methods added
  update UG with screenshot for add
  add pics for add function UG
  Reduce size of Manager UML image
  Update appt and apptList UML images to reflect new methods
  add line break
  crop picture
  fix findpatient exit error
  help-change
  add findappointment screenshot

# Conflicts:
#	docs/UserGuide.md

Anvitha-r committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T09-3/tp into UG-update-changes

* 'master' of https://github.com/AY2122S2-CS2113T-T09-3/tp:
  Fixing issue with loading data of older date

Anvitha-r committed:
Merge branch 'master' into UG-update-changes

* master:

# Conflicts:
#	docs/UserGuide.md

Anvitha-r committed:
update ug with screenshot

Anvitha-r committed:
fix checkstyle

shxr3f committed:
Merge pull request #206 from Anvitha-r/UG-update-changes

add missing images

tanweili committed:
Merge branch 'AY2122S2-CS2113T-T09-3:master' into master

tanweili committed:
Fix UG find section don't have hyperlink and remove some sections

shxr3f committed:
Merge pull request #209 from tanweili/WeiLi-FixUG

Wei li fix ug

tanweili committed:
Add sharif photo in and one more UG section hyperlink

Anvitha-r committed:
Merge pull request #210 from tanweili/WeiLi-FixUG

Add sharif photo in and one more UG section hyperlink

tanweili committed:
Hopefully this IS the final changes

shxr3f committed:
Merge pull request #211 from tanweili/WeiLiFFFFixes

Hopefully this IS the final changes

tanweili committed:
Change DOA to DOR in UG

tanweili committed:
Merge pull request #212 from tanweili/WeiLi-ChangeDOAtoDOR

Change DOA to DOR in UG

tanweili committed:
Update cmd summary add patient & doctor example ages

Age must corresponse with birthdate and in this case should be 22
